apps/api/src/app/blueprint/usecases/get-grouped-blueprints/get-grouped-blueprints.usecase.ts

```diff
@@ -37,27 +37,27 @@ export class GetGroupedBlueprints {
     return groups;
   }
 
-  private extractStoredBlueprints(groups: { name: string; blueprints: NotificationTemplateEntity[] }[]) {
-    return groups
-      .map((group) => group.blueprints)
-      .reduce((acc, curr) => {
-        acc.push(...curr);
-
-        return acc;
-      });
+  private groupedToBlueprintsArray(groups: { name: string; blueprints: NotificationTemplateEntity[] }[]) {
+    return groups.map((group) => group.blueprints).flat();
   }
 
   private updatePopularBlueprints(
     groups: { name: string; blueprints: NotificationTemplateEntity[] }[]
   ): INotificationTemplate[] {
-    const storedBlueprints = this.extractStoredBlueprints(groups);
+    const storedBlueprints = this.groupedToBlueprintsArray(groups);
+
+    const localPopularIds = [...POPULAR_TEMPLATES_ID_LIST];
 
-    const localPopularBlueprints = { ...POPULAR_TEMPLATES_GROUPED };
+    return localPopularIds.map((localBlueprintId) => {
+      const storedBlueprint = storedBlueprints.find((blueprint) => blueprint._id === localBlueprintId);
 
-    return localPopularBlueprints.blueprints.map((localBlueprint) => {
-      const storedBlueprint = storedBlueprints.find((blueprint) => blueprint._id === localBlueprint._id);
+      if (!storedBlueprint) {
+        Logger.warn(
+          `Could not find stored popular blueprint id: ${localBlueprintId}, BLUEPRINT_CREATOR: ${NotificationTemplateRepository.getBlueprintOrganizationId()}`
+        );
+      }
 
-      return (storedBlueprint ? storedBlueprint : localBlueprint) as INotificationTemplate;
+      return storedBlueprint as INotificationTemplate;
```

Comment:
So an empty array is ok to return?

Reply:
It should not happen, but it is possible 😅